helpers/parse_helpers.go

```diff
@@ -39,28 +47,44 @@ func processRecipientKeys(recipients []string) ([][]byte, [][]byte, [][]byte, er
 		case "jwe":
 			tmp, err := ioutil.ReadFile(value)
 			if err != nil {
-				return nil, nil, nil, errors.Wrap(err, "Unable to read file")
+				return nil, nil, nil, nil, nil, errors.Wrap(err, "Unable to read file")
 			}
 			if !encutils.IsPublicKey(tmp) {
-				return nil, nil, nil, errors.New("File provided is not a public key")
+				return nil, nil, nil, nil, nil, errors.New("File provided is not a public key")
 			}
 			pubkeys = append(pubkeys, tmp)
 
 		case "pkcs7":
 			tmp, err := ioutil.ReadFile(value)
 			if err != nil {
-				return nil, nil, nil, errors.Wrap(err, "Unable to read file")
+				return nil, nil, nil, nil, nil, errors.Wrap(err, "Unable to read file")
 			}
 			if !encutils.IsCertificate(tmp) {
-				return nil, nil, nil, errors.New("File provided is not an x509 cert")
+				return nil, nil, nil, nil, nil, errors.New("File provided is not an x509 cert")
 			}
 			x509s = append(x509s, tmp)
 
+		case "pkcs11":
+			tmp, err := ioutil.ReadFile(value)
+			if err != nil {
+				return nil, nil, nil, nil, nil, errors.Wrap(err, "Unable to read file")
+			}
+			if !encutils.IsPkcs11PublicKey(tmp) {
+				if !encutils.IsPublicKey(tmp) {
+					return nil, nil, nil, nil, nil,
+						errors.New("Provided file is not a public key")
+				} else {
+					pkcs11Pubkeys = append(pkcs11Pubkeys, tmp)
+				}
+			} else {
+				pkcs11Yamls = append(pkcs11Yamls, tmp)
+			}
```

Comment:
nit suggestion: may help readability with flow (similar to what you did with private keys)
```
if IsPKCS11PublicKey {
   ...
} else if IsPublicKey {
  ...
} else {
 err
}
```